benchmarks/bench_ml.py

```diff
@@ -0,0 +1,422 @@
+    # OUTPUT_TYPE:  OUTPUT_CONTINUOUS / OUTPUT_CATEGORY
+    # MODEL_PARAMETERS: dictionary whose keys correspond to parameters that are passed to __init__ of python class
+    #                   that implements model
+    # MODEL_BACKEND: python class that implements model
+    models = {
+        #
+        #   regressors go here
+        #
+        MLPRegressor.__name__:{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            MODEL_PARAMETERS: nnparams,
+            MODEL_BACKEND: MLPRegressor,
+        },
+        SVR.__name__:{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            MODEL_PARAMETERS: {
+                'C': (Real(-4.0, 4.0), pow10map),
+                'epsilon': (Real(-4.0, 1.0), pow10map),
+                'gamma': (Real(-4.0, 1.0), pow10map),
+            },
+            MODEL_BACKEND: SVR,
+        },
+        DecisionTreeRegressor.__name__:{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            MODEL_PARAMETERS: {
+                'max_depth': (Real(1.0, 4.0), pow2intmap),
+                'min_samples_split': (Real(1.0, 8.0), pow2intmap),
+            },
+            MODEL_BACKEND: DecisionTreeRegressor,
+        },
+        #
+        #   classifiers go here
+        #
+        MLPClassifier.__name__:{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            MODEL_PARAMETERS: nnparams,
+            MODEL_BACKEND: MLPClassifier,
+        },
+        SVC.__name__:{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            MODEL_PARAMETERS: {
+                'C': (Real(-4.0, 4.0), pow10map),
+                'gamma': (Real(-4.0, 1.0), pow10map),
+            },
+            MODEL_BACKEND: SVC,
+        },
+        DecisionTreeClassifier.__name__: {
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            MODEL_PARAMETERS: {
+                'max_depth': (Real(1.0, 4.0), pow2intmap),
+                'min_samples_split': (Real(1.0, 8.0), pow2intmap),
+            },
+            MODEL_BACKEND: DecisionTreeClassifier,
+        },
+    }
+
+    # every dataset should have fields:
+    # OUTPUT_TYPE:  OUTPUT_CONTINUOUS / OUTPUT_CATEGORY
+    # DATASET_LOADER: callable that loads data into matrix X of inputs and vector y of outputs
+    datasets={
+        "Boston":{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            DATASET_LOADER: lambda : load_data_target(skd.load_boston())
+        },
+        "Housing":{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_california_housing())
+        },
+        #
+        #   classification datasets go here
+        #
+        "Leukemia":{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('leukemia'))
+        },
+        "Climate Model Crashes":{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('climate-model-simulation-crashes', target_name=-1))
+        },
+    }
```

Comment:
Currently those are static variables for the class. Do you think it would be better if I would move them outside?